File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> contain;
        int n=nums.size();
        for(int i=0; i<n; i++){
            contain[nums[i]]=i;
        }

        for(int i=0; i<n; i++){
            int desired=target-nums[i];
            if(contain.find(desired)!=contain.end() && contain[desired]!=i){
                return {i, contain[desired]};
            }
        }
        return {};
    }
};

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hm ={}
        for i,num in enumerate(nums):
            req=target-num
            if(req in hm):
                return[hm[req],i]
            hm[num]=i

